### PR TITLE
Add test plan for `sycl_ext_oneapi_enqueue_functions` extension

### PR DESCRIPTION
**Pull request by 0x12CC**

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
It would be good to also have `static_assert` on the return type of all of these functions, but other than that I think this looks reasonable.

File: test_plans/sycl_ext_oneapi_enqueue_functions.asciidoc [hunk 1]
```diff
@@ -0,0 +1,219 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for sycl_ext_oneapi_enqueue_functions
+
+This is a test plan for the API described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_enqueue_functions.asciidoc[sycl_ext_oneapi_enqueue_functions].
```
Comment (keryell) [inline test_plans/sycl_ext_oneapi_enqueue_functions.asciidoc:7]: File not found.

Comment (steffenlarsen) [reply test_plans/sycl_ext_oneapi_enqueue_functions.asciidoc:7]: Good eye. It was recently moved to
```suggestion
https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc[sycl_ext_oneapi_enqueue_functions].
```

Comment (bader) [reply test_plans/sycl_ext_oneapi_enqueue_functions.asciidoc]: I suggest we use permanent link here.

**Comment by bader** [comment]
@gmlueck, ping.

**Comment by gmlueck** [comment]
> @gmlueck, ping.

I trust @steffenlarsen's review.  Is there a specific question you would like me to answer?

**Comment by bader** [comment]
> > @gmlueck, ping.
> 
> I trust @steffenlarsen's review. Is there a specific question you would like me to answer?

You are the code owner of test plans for extensions, so formally either you or SYCL implementation representatives (i.e. @KhronosGroup/sycl-cts-reviewers team) have to approve test plans.
I'm really appreciate the efforts @steffenlarsen is making to push SYCL-CTS PRs forward, so I suggest we make him a formal reviewer of SYCL-CTS code and/or test plans.

**Comment by gmlueck** [comment]
> I'm really appreciate the efforts @steffenlarsen is making to push SYCL-CTS PRs forward, so I suggest we make him a formal reviewer of SYCL-CTS code and/or test plans.

Yes, I agree.

**Comment by keryell** [comment]
@gmlueck It could be interesting if these functions were able to be found via ADL to avoid the namespace, for example if  the extension header is used.

**Review by keryell — APPROVED** [review]
That looks good.

**Comment by gmlueck** [comment]
> @gmlueck It could be interesting if these functions were able to be found via ADL to avoid the namespace, for example if the extension header is used.

In order to do this, I think we would need to define these functions in the `sycl` namespace instead of the `sycl::ext::oneapi` namespace.  However, this would be contrary to our goal of exposing extended APIs only via an "extension" namespace.  Therefore, we would probably not do this.  However, if this extension was adopted into a future version of the core spec, then it would be defined in the `sycl` namespace, and it would therefore be available via ADL.

**Pull request merged**